manifests/volume.pp

```diff
@@ -222,12 +222,24 @@
       }
 
       # did the options change?
-      $current_options = getvar("gluster_volume_${title}_options")
-      if $current_options {
-        $_current = sort( split($current_options, ',') )
-      } else {
-        $_current = []
+      if (defined('$::gluster_volume_options')){
+          # Use the new structured fact if it exists.
+          if (has_key($::gluster_volume_options, $title)){
+            $_current = sort($::gluster_volume_options[$title])
+          } else {
+            $_current = []
+          }
+      }else{
+          # Use the legacy facts.
```

Comment:
I'm wondering if this needs to be supported. AFAIK facts are always updated on every run so we should be able to rely on the new one.

Reply:
I wasn't sure what would happen on older puppet versions.  I don't have anything older than 5 in my lap to test my code.  I'm happy to remove it as extraneous.  I'm also wondering if we want to make the structured fact more complex to have everything we know about a volume in a gluster_volumes Hash that includes bricks, ports, and options.  I sadly don't have the time to rewrite all of the other code to use the new fact structure but would happily make the fact change so that someone else can take it from there (and support the options part as I need that to work for my current production environment)